internal/venice-common/src/main/java/com/linkedin/venice/kafka/TopicManager.java

```diff
@@ -781,9 +783,10 @@ public List<PartitionInfo> partitionsFor(String topic) {
 
   /**
    * @deprecated this is only used by {@link #isTopicFullyDeleted(String, boolean)} in cases where the
-   *             {@link com.linkedin.venice.kafka.admin.ScalaAdminUtils} is used. We should deprecate
+   *             ScalaAdminUtils is used. We should deprecate
    *             both the Scala admin as well as this function, so please do not proliferate its usage
    *             in new code paths.
+   * TODO: remove the usage of this raw consumer.
```

Comment:
So there are more usages then?

We should not refer to `ScalaAdminClient` in the JavaDoc if that's not part of the code base anymore...

Reply:
There are three tech debts I found:
1. This code piece you highlighted here: we have an additional check to see whether the topic is really deleted after topic deletion request future completes.
2. We built two TopicManager in servers, one is configurable, the other is always using JavaAdminClient
3. We introduced different configs, like admin client class, read/write-only admin client class.

My original plan was to upgrade first, and merge it in today/this week's release; and clean up the tech debts later, since there is a committed timeline to catch up on. Will take a stab at cleaning these tech debts tomorrow.

Reply:
Ok